### PR TITLE
#170947618 view new Notifications.

### PR DESCRIPTION
**Pull request by Niyonsengaeric**
#### What does this PR do?
- allow a user to view his/her unread notifications.
#### Description of Task to be completed?
- allow a user to view his/her new notifications.
#### How should this be manually tested?
- run `git clone https://github.com/Stackup-Rwanda/knights-bn-frontend.git` to clone the repo.
- run `cd knights-bn-backend` in your terminal to enter into the repository.
- run `git fetch ` command to download commits, files, and refs from a remote repository into your local repo.
- run `git checkout ft-approved-notification-170947618` to switch to this branch.
- run `npm install` to install dependencies.
- run `npm run dev-start` to start the application.
  you will see the login page then login with your email or password 
```
        "email":"alain.maxime@gmail.com",
	"password":"Niyonkuru@1"
```
- use `http://localhost:8080/request` to create a request.

- log in again as an administrator.
```
administrator user:

        "email":"william.ishimwe@andela.com",
	"password":"Password@1"
```
- Go to pending requests then approve your request.
- pending request URL: - use `http://localhost:8080/pending` to create a request.

- login again into your account you will see the notification of the approved request.

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
N/A
#### Screenshots (if appropriate)

      new notifications

Screen Shot 2020-06-03 at 10 22 34

    see notifications

Screen Shot 2020-06-03 at 10 22 42

### CODE REVIEW

**Review by Kagorora — APPROVED** [review]

**Review by T2Wil — APPROVED** [review]
Nice work @Niyonsengaeric .

File: __tests__/redux/action/action.test.js [hunk 1]
```diff
@@ -305,4 +311,52 @@ describe(' test loginUsersSuccess actions', () => {
...
+          updatedAt: '2020-05-26T16:58:20.381Z',
+        },
+      ],
+    };
+    expect(
+      getNotifications({
+        type: 'GET_NOTIFICATION',
+        requests,
+      }).payload.requests.message,
+    ).toEqual('Your notifications');
```
Comment (ceelogre) [inline __tests__/redux/action/action.test.js:378]: What are you testing for Exactly here?

File: __tests__/redux/action/action.test.js [hunk 2]
```diff
@@ -305,4 +311,52 @@ describe(' test loginUsersSuccess actions', () => {
...
+    const response = {
+      isLoggedIn: true,
+      message: 'You have no unread notification',
+    };
+    expect(
+      markAllAsRead({
+        type: 'MARK_READ',
+        response,
+      }).message.response.message,
+    ).toEqual('You have no unread notification');
```
Comment (ceelogre) [inline __tests__/redux/action/action.test.js:390]: Same comment as previous.

File: __tests__/redux/action/action.test.js (hunk 1 above)
Comment (Niyonsengaeric) [reply __tests__/redux/action/action.test.js:378]: @ceelogre here I am testing when the notifications are successfully returned what will be the message that should be displayed

Comment (ceelogre) [reply __tests__/redux/action/action.test.js:378]: But you set the message previously in the variable.

Comment (Niyonsengaeric) [reply __tests__/redux/action/action.test.js:378]: @ceelogre I was trying to test the actions by passing  variables.

File: README.md [hunk 3]
```diff
@@ -13,32 +13,33 @@ Make company global travel and accommodation easy and convenient for the strong
...
 ## Technologies used :gear:
 - React JS
+- Bootstrap
 - Webpack
 - Redux
 - SASS
 - Node JS
 - Jest
-- React-testing-library
+- Enzyme
```
Comment (kagabof) [inline README.md:23]: Please add
- Circle ci
- Code-climate
- Heroku
- ESLint
- PivotalTracker 
- Github
etc..

File: src/components/Nav.js [hunk 4]
```diff
@@ -1,23 +1,44 @@
...
 import logo from '../assets/images/logo@2x.png';
 import Languages from './Languages';
 import notificationIcon from '../assets/images/notification.svg';
 import AddProfile from './AddProfile';
+import { getNotifications, markAllAsRead } from '../redux/actions/index';
+import convertDate from '../util/convertDate';
 
 env.config();
-const Nav = () => {
+const Nav = (props) => {
```
Comment (kagabof) [inline src/components/Nav.js]: Please validate your props in the component.

**Review by kagabof — CHANGES_REQUESTED** [review]

**Review by kagabof — APPROVED** [review]

**Review by kagabof — APPROVED** [review]

**Pull request merged**